app/src/main/java/ch/epfl/skysync/viewmodel/FlightsViewModel.kt

```diff
@@ -45,21 +53,37 @@ class FlightsViewModel(
   private val _currentFlightTypes: MutableStateFlow<List<FlightType>> =
       MutableStateFlow(emptyList())
   private val _currentVehicles: MutableStateFlow<List<Vehicle>> = MutableStateFlow(emptyList())
+  private val _currentUser = MutableStateFlow<User?>(null)
 
   val currentFlights = _currentFlights.asStateFlow()
   val currentBalloons = _currentBalloons.asStateFlow()
   val currentBaskets = _currentBaskets.asStateFlow()
   val currentFlightTypes = _currentFlightTypes.asStateFlow()
   val currentVehicles = _currentVehicles.asStateFlow()
+  val currentUser = _currentUser.asStateFlow()
 
   fun refresh() {
-    refreshCurrentFlights()
+    refreshUserAndFlights()
     refreshCurrentBalloons()
     refreshCurrentBaskets()
     refreshCurrentFlightTypes()
     refreshCurrentVehicles()
   }
 
+  fun refreshUserAndFlights() =
+      viewModelScope.launch {
+        _currentUser.value = repository.userTable.get(userId ?: UNSET_ID, onError = { onError(it) })
```

Comment:
You should not try to fetch a user with an unset id, it will throw an error, instead you should require to have an not null uid in your view model (which will always be the case anyway), so you could have at line 32: ’val uid: String,’

Reply:
got it, thanks